mlflow/types/llm.py

```diff
@@ -337,16 +365,7 @@ def __post_init__(self):
         self._convert_dataclass_list("choices", ChatChoice)
         if isinstance(self.usage, dict):
             self.usage = TokenUsageStats(**self.usage)
-        if not isinstance(self.usage, TokenUsageStats):
-            raise ValueError(
-                f"Expected `usage` to be of type TokenUsageStats or dict, got {type(self.usage)}"
-            )
-
-    def __setattr__(self, name, value):
-        # A hack to ensure users cannot overwrite 'object' field
-        if name == "object" and value != "chat.completion":
-            raise ValueError("`object` field must be 'chat.completion'")
-        return super().__setattr__(name, value)
+        self._validate_field("usage", TokenUsageStats, False)
```

Comment:
not adding validation for `metadata`, so that users can have more flexibility. however, i'm not sure if this is a pattern we want to encourage, so in the documentation the type is still `Dict[str, str]`, matching the model signature.